src/main/java/core/basesyntax/SupplyOperation.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+import java.util.Map;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void apply(Map<String, Integer> storage, FruitTransaction transaction) {
```

Comment:
Please ensure that the `apply` method in `SupplyOperation` adheres to the encapsulation principle. If this method is not intended to be used outside of the package or subclasses, consider reducing its visibility.